pyodide_build/tests/test_common.py

```diff
@@ -158,32 +161,80 @@ def test_check_wasm_magic_number(tmp_path):
     assert check_wasm_magic_number(tmp_path / "badfile.so") is False
 
 
-def test_default_xbuildenv_path(tmp_path, reset_cache):
-    import platformdirs
+class MockConfigManager:
```

Comment:
If `get_host_build_flag` is used (from my previous comment), we won't need to the mock config manager anymore. We can mock the return value of `get_host_build_flag` instead.

Also, the functionality of the config manager is tested separately in `test_config.py`, so I don't think we need multiple cases to test the precedence of env var and config file here.